pydicom/tests/test_gdcm_pixel_data.py

```diff
@@ -63,18 +71,27 @@
 save_dir = os.getcwd()
 
 
-@pytest.mark.skipif(have_gdcm_handler, reason=gdcm_present_message)
 class GDCM_JPEG_LS_Tests_no_gdcm(unittest.TestCase):
     def setUp(self):
-        self.jpeg_ls_lossless = read_file(jpeg_ls_lossless_name)
+        self.utf8_filename = os.path.join(tempfile.gettempdir(), "ДИКОМ.dcm")
+        shutil.copyfile(jpeg_ls_lossless_name, self.utf8_filename)
+        try:
+            self.jpeg_ls_lossless = read_file(
+                unicode(self.utf8_filename,
+                        encoding=sys.getfilesystemencoding()))
+        except NameError:
```

Comment:
Python's unicode/string mess just baffles me sometimes.  I changed this to "utf8" in one of those commits below, and it broke. With sys.getfilesystemencoding(), I think it just worked by accident (it would not write the intended file, but it was still a valid filename). So then I redid it such that all the paths for this test are unicode. That, I think, is probably the "correct" way to do it, but it seems a bit of juggling to just write a fancy filename.